src/components/molecules/VenueChat/VenueChat.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext, useEffect, useState, useMemo, FC } from "react";
+import { ChatContext } from "components/context/ChatContext";
+import "./CampChat.scss";
+import { useUser } from "hooks/useUser";
+import { useSelector } from "hooks/useSelector";
+import { useVenueId } from "hooks/useVenueId";
+import { getDaysAgoInSeconds } from "utils/time";
+import { VENUE_CHAT_AGE_DAYS } from "settings";
+import ChatBox from "components/molecules/Chatbox";
+import useRoles from "hooks/useRoles";
+
+interface ChatOutDataType {
+  messageToTheBand: string;
+}
+
+const VenueChat: FC = () => {
+  const venueId = useVenueId();
+  const { userRoles } = useRoles();
+  const { user } = useUser();
+  const { chats, venue } = useSelector((state) => ({
+    chats: state.firestore.ordered.venueChats,
+    venue: state.firestore.data.currentVenue,
+  }));
+
+  const [isMessageToTheBarSent, setIsMessageToTheBarSent] = useState(false);
+
+  useEffect(() => {
+    if (isMessageToTheBarSent) {
+      setTimeout(() => {
+        setIsMessageToTheBarSent(false);
+      }, 2000);
+    }
+  }, [isMessageToTheBarSent]);
+
+  const chatContext = useContext(ChatContext);
+
+  const onMessageSubmit = async (data: ChatOutDataType) => {
+    chatContext &&
+      user &&
+      chatContext.sendRoomChat(user.uid, venueId!, data.messageToTheBand);
+  };
+
+  function roundToNearestHour(seconds: number) {
+    const oneHour = 60 * 60;
+    return Math.floor(seconds / oneHour) * oneHour;
+  }
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      chats &&
+      chats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "room" &&
+            message.to === venueId &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
+    [chats, venueId, HIDE_BEFORE]
+  );
```

Comment:
[nit] A nit here would potentially be extracting the `filter`/`sort` functions.. but in this case I think they may be small/easy/obvious enough that it makes sense to use here, particularly as it's inside of a `useMemo` as well.

Though if the `filter`/`sort` functions are generic enough that we may want to use them again somewhere in future, I would still preference extracting them (the `sort` in particular, in this case)

Reply:
nit, memo'd, fairly specific filter. resolving this